### PR TITLE
Feat/werzeug sebastian

### PR DESCRIPTION
**Pull request by Biter01**
Added:
new package tools
 -Brush.java 
 -WorkingTool.java

 new package dialog
    -DrawDialog.java

You can now select a color and a brush size and paint with the brush. 
Working Area has now the possibiltiy to execute every new Tool which has implemented the new WorkingTool interface

### CODE REVIEW

**Comment by Biter01** [comment]
Added the new Files

**Comment by VanDulti** [comment]
Strg + Alt + L in allen Files

File: pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/WorkingArea.java [hunk 1]
```diff
@@ -137,36 +145,73 @@ public void componentResized(ComponentEvent e) {
...
+
+	/*
+		ToolListener executes Tools on a certain Mouse Event
+		Tools are executed with tool.execute() for example Brush Tool
+	 */
+	private class ToolListener extends MouseAdapter {
 		private Point initialPoint = null;
 
+		//All tools that will be executed are saved here
+		private static WorkingTool tool = null;
```
Comment (VanDulti) [inline pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/WorkingArea.java:165]: static 🥶

File: pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/WorkingArea.java [hunk 2]
```diff
@@ -137,36 +145,73 @@ public void componentResized(ComponentEvent e) {
...
 
+		private int getRelativeX(Point point) {
+			return (int)((point.x -x)/scale);
+		}
+
+		private int getRelativeY(Point point) {
+			return (int)((point.y -y)/scale);
+		}
+		public static void setTool(WorkingTool toolee) {
+			tool= toolee;
```
Comment (VanDulti) [inline pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/WorkingArea.java:212]: 😳

File: pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/WorkingArea.java [hunk 3]
```diff
@@ -202,6 +244,8 @@ public void mouseDragged(MouseEvent e) {
 			restrictBounds();
 
 			initialPoint = currentPoint;
+			new Rectangle();
```
Comment (VanDulti) [inline pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/WorkingArea.java:247]: ...

**Review by VanDulti — CHANGES_REQUESTED** [review]

**Review by VanDulti — APPROVED** [review]

**Pull request merged**